codegen/src/generator/mod.rs

```diff
@@ -62,7 +65,15 @@ where P: GenerateProtocol {
         }}
         ",
         methods = protocol_generator.generate_methods(service),
-        service_name = &service.metadata.service_abbreviation,
+        service_name = match &service.metadata.service_abbreviation {
+            &Some(ref service_abbreviation) => service_abbreviation.as_str(),
+            &None => {
+                match service.metadata.endpoint_prefix {
```

Comment:
@matthewkmayer do you want me to take a look at what impacts the change may have or are you going to handle it?